frontend/packages/topology/src/components/export-app/ExportApplicationModal.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import { ActionGroup, Button } from '@patternfly/react-core';
+import { useTranslation, Trans } from 'react-i18next';
+import {
+  ModalTitle,
+  ModalBody,
+  ModalComponentProps,
+  ModalFooter,
+} from '@console/internal/components/factory/modal';
+
+export type ExportApplicationModalProps = ModalComponentProps & {
+  namespace: string;
+  startTime?: string;
+  onCancelExport?: () => Promise<boolean>;
+  onRestartExport?: () => Promise<boolean>;
+};
+
+const ExportApplicationModal: React.FC<ExportApplicationModalProps> = (props) => {
+  const { t } = useTranslation();
+  const { cancel, namespace, startTime, onCancelExport, onRestartExport } = props;
+  const [errMessage, setErrMessage] = React.useState<string>('');
+
+  const onCancel = async () => {
+    try {
+      await onCancelExport();
```

Comment:
since this is an optional prop
```suggestion
      await onCancelExport?.();
```

Reply:
onCancel will be invoked only if `onCancelExport` is present as btn is rendered then